OWML.ModLoader/ModSorter.cs

```diff
@@ -0,0 +1,105 @@
+﻿using OWML.Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OWML.ModLoader
+{
+    class ModSorter
+    {
+        public static IList<ModDep> SortMods(IList<IModData> mods)
+        {
+            List<ModDep> modList = new List<ModDep>();
+            foreach (var mod in mods)
+            {
+                if (mod.Manifest.Dependencies == null)
+                {
+                    mod.Manifest.SetDependencies(new[] { "None" });
+                }
+
+                if (mod.Manifest.Dependencies.First() != "None")
+                {
+                    modList.Add(new ModDep(mod.Manifest.Name, mod, mod.Manifest.Dependencies));
+                }
+                else
+                {
+                    modList.Add(new ModDep(mod.Manifest.Name, mod));
+                }
+            }
+
+            return Sort(modList, x => x.Dependencies, x => x.Name);
+        }
+
+        private static IList<T> Sort<T>(IEnumerable<T> source, Func<T, IEnumerable<T>> getDependencies)
+        {
+            var sorted = new List<T>();
+            var visited = new Dictionary<T, bool>();
+
+            foreach (var item in source)
+            {
+                Visit(item, getDependencies, sorted, visited);
+            }
+
+            return sorted;
+        }
+
+        private static void Visit<T>(T item, Func<T, IEnumerable<T>> getDependencies, List<T> sorted, Dictionary<T, bool> visited)
```

Comment:
What does is "Visit" mean in this context?

Reply:
to look at an item